snippets/current/src/register-schema-and-cred-def.ts

```diff
@@ -0,0 +1,134 @@
+import { DidsModule, InitConfig, KeyType, LogLevel, TypedArrayEncoder } from '@aries-framework/core'
+
+const config: InitConfig = {
+  label: 'docs-agent-nodejs-register-schema-and-cred-def',
+  walletConfig: {
+    id: 'wallet-id',
+    key: 'testkey0000000000000000000000000',
+  },
+}
+
+// start-section-1
+import { Agent } from '@aries-framework/core'
+import { agentDependencies } from '@aries-framework/node'
+import { AskarModule } from '@aries-framework/askar'
+import { ariesAskar } from '@hyperledger/aries-askar-nodejs'
+import {
+  IndyVdrAnonCredsRegistry,
+  IndyVdrIndyDidRegistrar,
+  IndyVdrIndyDidResolver,
+  IndyVdrModule,
+} from '@aries-framework/indy-vdr'
+import { indyVdr } from '@hyperledger/indy-vdr-nodejs'
+import { AnonCredsModule } from '@aries-framework/anoncreds'
+import { AnonCredsRsModule } from '@aries-framework/anoncreds-rs'
+import { anoncreds } from '@hyperledger/anoncreds-nodejs'
+
+const agent = new Agent({
+  config,
+  dependencies: agentDependencies,
+  modules: {
+    // Register the Askar module on the agent
+    // We do this to have access to a wallet
+    askar: new AskarModule({
+      ariesAskar,
+    }),
+    anoncredsRs: new AnonCredsRsModule({
+      anoncreds,
+    }),
+    indyVdr: new IndyVdrModule({
+      indyVdr,
+      networks: [
+        {
+          isProduction: false,
+          indyNamespace: 'bcovrin:test',
+          genesisTransactions: '<genesis transactions>',
+          connectOnStartup: true,
+        },
+      ],
+    }),
+    anoncreds: new AnonCredsModule({
+      registries: [new IndyVdrAnonCredsRegistry()],
+    }),
+    dids: new DidsModule({
+      registrars: [new IndyVdrIndyDidRegistrar()],
+      resolvers: [new IndyVdrIndyDidResolver()],
+    }),
+  },
+})
+// end-section-1
+
+// start-section-2
+const registerDid = async () => {
+  await agent.dids.import({
+    did: 'did:indy:bcovrin:test:MqbfZSYXoeuLfgJne6gQVK',
+    overwrite: true,
+    privateKeys: [
+      {
+        privateKey: TypedArrayEncoder.fromString('secret-docs-seed0000000000000000'),
+        keyType: KeyType.Ed25519,
+      },
+    ],
+  })
+}
+// end-section-2
+
+// start-section-3
+const registerSchema = async () => {
```

Comment:
As mentioned before, I think we should move away from these methods, as it makes the snippet in general more complex when just going through the tutorial. 

I made a PR against your branch to support top-level await and added an example of how it would like with top level await for this file: https://github.com/blu3beri/aries-javascript-docs/pull/1